### PR TITLE
Handle SSL errors with retries

### PR DESCRIPTION
**Pull request by rryk**
This is a re-created pull request #159 to make sure that commit author and pull requests are the same accounts so that CLA bot is not confused. This is going to address issues #71, #97, #158.

### CODE REVIEW

File: tests/test_http.py [hunk 1]
```diff
@@ -593,6 +622,36 @@ def test_unicode(self):
...
+        HttpMockWithSSLErrors(3, {'status': '200'}, '{"foo": "bar"}'),
+        model.response,
+        u'https://www.example.com/json_api_endpoint')
+    request._sleep = lambda _x: 0  # do nothing
+    request._rand = lambda: 10
+    response = request.execute(num_retries=3)
+    self.assertEqual({u'foo': u'bar'}, response)
+
+  def test_retry_ssl_errors_resumable(self):
+    f = open(datafile('small.png'), 'rb')
```
Comment (nathanielmanistaatgoogle) [inline tests/test_http.py]: Does f not need to be closed? What would you think of writing these two lines as
```
with open(datafile('small.png'), 'rb') as small_png_file:
  small_png_file_descriptor = BytesIO(small_png_file.read())
```
?

(I also happen to dislike initialisms and abbreviations for identifiers.)

**Comment by nathanielmanistaatgoogle** [comment]
Sorry for the delay on this latest round of review.

File: tests/test_http.py (hunk 1 above)
Comment (sergiyby) [inline tests/test_http.py]: Done.

**Comment by nathanielmanistaatgoogle** [comment]
Content looks good but please squash commits - there's no need for typo-correction-during-code-review to be its own unique part of project history. :-P

**Comment by rryk** [comment]
Done.

**Comment by rryk** [comment]
Done.

**Comment by nathanielmanistaatgoogle** [comment]
Thanks for the contribution!

**Pull request merged**